src/Microsoft.SqlTools.ServiceLayer/Formatter/TSqlFormatterService.cs

```diff
@@ -103,6 +110,14 @@ private static TelemetryProperties CreateTelemetryProps(bool isDocFormat)
 
         private async Task<TextEdit[]> FormatRangeAndReturnEdits(DocumentRangeFormattingParams docFormatParams)
         {
+            if (docFormatParams == null
+                || docFormatParams.TextDocument == null
+                || docFormatParams.TextDocument.Uri == null
+                || ShouldSkipFormatting(docFormatParams.TextDocument.Uri))
+            {
+                return Array.Empty<TextEdit>();
+            }
+
             return await Task.Factory.StartNew(() =>
```

Comment:
What's the benefit of starting up a new thread for something that returns immediately? Why not do a `Task.FromResult()` sort of thing?

Reply:
I'll put the other code inside the taskfactory, but the purpose is that these are event handlers for message channel notifications. Per previous discussions we don't want to hold these up - formatting the code is usually fast but for a large file could take a second. Why hold the channel up for that?